modVM/src/org/aion/vm/TransactionExecutor.java

```diff
@@ -50,7 +51,7 @@
  * @author yulong
  */
 public class TransactionExecutor extends AbstractExecutor {
-
+    // provider is essential to execute, but it is only set manually with a setter, not good practice
```

Comment:
should we just add it as parameter to the constructor?  (if it takes too long to refactor the existing calls to this constructor, we just change all the existing calls to the constructor to pass in null, that's should end up being the exact same behaviour)

Reply:
maybe this is almost worth a separate PR. I'm not sure what the motivation behind this was, it may have a use. I think at least we could instantiate the provider so that if someone forgets to explicitly set it nothing crashes

Reply:
Sounds fair ... it's not really a big deal for me, but if not fixed now, it'll probably be a while before anyone else notices it again :)